src/Linters/PackagePrivateLinter.hack

```diff
@@ -0,0 +1,318 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+use type Facebook\HHAST\{
+  ASTLintError,
+  ASTLinter,
+  FunctionToken,
+  INamespaceUseDeclaration,
+  NameToken,
+  NamespaceDeclaration,
+  NamespaceGroupUseDeclaration,
+  NamespaceToken,
+  NamespaceUseClause,
+  NamespaceUseDeclaration,
+  Node,
+  QualifiedName,
+  Script,
+  TypeToken,
+};
+use namespace HH\Lib\{C, Str, Vec};
+
+type use_definition_parts_t = shape(
```

Comment:
- `type TUseDefinitionParts` would be following the convention here
- these should probably be type constants on the class instead of top-level types in the HHAST namespace